### PR TITLE
Added training plan logic into automate script

### PR DESCRIPTION
**Pull request by amanshanbhag**
New feature (training plans) integrated into the auto create cluster script. Also updated README with better descriptions + added in gif of demo :)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by nghtm** [comment]
please resolve conflicts before merging:

https://github.com/aws-samples/awsome-distributed-training/pull/510/conflicts

**Comment by amanshanbhag** [comment]
It's been tested pre-release. Waiting on account AL to be able to test once before merging.

File: 1.architectures/5.sagemaker-hyperpod/automate-smhp-slurm/README.md [hunk 1]
```diff
@@ -1,9 +1,94 @@
-# Automate SageMaker HyperPod Cluster Creation
-A bash script that automates the manual cluster creation process for SageMaker HyperPod SLURM  
+# SageMaker Hyperpod Cluster Automation Script
 
-This automates the steps from the [SageMaker HyperPod SLURM Workshop](https://catalog.workshops.aws/sagemaker-hyperpod/en-US)
+This project provides a script to automate the creation and setup of a SageMaker Hyperpod cluster with SLURM integration.
 
-## 🚀 Installation and Usage
-Using this script is very simple. Run ```bash automate-cluster-creation.sh```
+The automation script streamlines the process of setting up a distributed training environment using AWS SageMaker Hyperpod.
```
Comment (mhuguesaws) [inline 1.architectures/5.sagemaker-hyperpod/automate-smhp-slurm/README.md:5]: redunddant with previous sentence.

File: 1.architectures/5.sagemaker-hyperpod/automate-smhp-slurm/README.md [hunk 2]
```diff
@@ -1,9 +1,94 @@
...
+It handles the installation and configuration of necessary tools, clones the required repository, sets up environment variables, and configures lifecycle scripts for the SageMaker Hyperpod architecture.
```
Comment (mhuguesaws) [inline 1.architectures/5.sagemaker-hyperpod/automate-smhp-slurm/README.md:6]:
```suggestion
It installs necessary tools, clones the required repository, sets up environment variables, and configures lifecycle scripts for the SageMaker Hyperpod architecture.
```

File: 1.architectures/5.sagemaker-hyperpod/automate-smhp-slurm/README.md [hunk 3]
```diff
@@ -1,9 +1,94 @@
...
+
+1. Check and install/update AWS CLI if necessary.
+2. Verify Git installation.
+3. Clone the "awsome-distributed-training" repository.
+4. Set up environment variables.
+5. Configure lifecycle scripts for SageMaker Hyperpod.
+
+### Configuration
+
+During execution, you'll be prompted to provide the following information:
```
Comment (mhuguesaws) [inline 1.architectures/5.sagemaker-hyperpod/automate-smhp-slurm/README.md:57]:
```suggestion
During execution, you will provide the following information:
```

File: 1.architectures/5.sagemaker-hyperpod/automate-smhp-slurm/README.md [hunk 4]
```diff
@@ -1,9 +1,94 @@
-# Automate SageMaker HyperPod Cluster Creation
-A bash script that automates the manual cluster creation process for SageMaker HyperPod SLURM  
+# SageMaker Hyperpod Cluster Automation Script
```
Comment (mhuguesaws) [inline 1.architectures/5.sagemaker-hyperpod/automate-smhp-slurm/README.md:1]:
```suggestion
# Amazon SageMaker AI Hyperpod Cluster Automation Script
```

File: 1.architectures/5.sagemaker-hyperpod/automate-smhp-slurm/README.md [hunk 5]
```diff
@@ -1,9 +1,94 @@
...
 
-This automates the steps from the [SageMaker HyperPod SLURM Workshop](https://catalog.workshops.aws/sagemaker-hyperpod/en-US)
+This project provides a script to automate the creation and setup of a SageMaker Hyperpod cluster with SLURM integration.
```
Comment (mhuguesaws) [inline 1.architectures/5.sagemaker-hyperpod/automate-smhp-slurm/README.md:3]:
```suggestion
This project provides a script to automate the creation and setup of a Amazon SageMaker Hyperpod cluster with SLURM integration.
```

**Review by mhuguesaws — COMMENTED** [review]
Left comments

**Comment by amanshanbhag** [comment]
Thank you Maxime! i will merge changes after a final test

**Comment by amanshanbhag** [comment]
Tested with account tp. Good to merge

**Review by sean-smith — APPROVED** [review]

**Pull request merged**